azure-pipelines.yml

```diff
@@ -291,6 +307,53 @@ jobs:
     workingDirectory: build
     displayName: 'Run Ctest'
 
+- job:
+  displayName: Win Intel
+  pool:
+    vmImage:  'windows-2019'
+  strategy:
+    matrix:
+      64-bit Debug ICL:
+        BuildType: Debug
+        CMakeArgs: '-GNinja -DCMAKE_C_COMPILER="icl" -DCMAKE_CXX_COMPILER="icl"'
+        JFlag: -j 4
+        CompilerURI: "https://registrationcenter-download.intel.com/akdlm/irc_nas/17762/w_HPCKit_p_2021.2.0.2901_offline.exe"
+        CompilerDir: 'C:\Program Files (x86)\Intel\oneAPI'
+      64-bit Release ICL:
+        BuildType: Release
+        CMakeArgs: '-GNinja -DCMAKE_C_COMPILER="icl" -DCMAKE_CXX_COMPILER="icl"'
+        JFlag: -j 4
+        CompilerURI: "https://registrationcenter-download.intel.com/akdlm/irc_nas/17762/w_HPCKit_p_2021.2.0.2901_offline.exe"
+        CompilerDir: 'C:\Program Files (x86)\Intel\oneAPI'
+  steps:
+    - script: |
+        bash -c "cat /proc/cpuinfo"
+        bash -c "cat /proc/meminfo"
+      displayName: 'Machine stats'
+
+    - script: |
+        choco install --accept-license -y Ninja curl
```

Comment:
What is the license that we're accepting?  If we are accepting an EULA on behalf of Microsoft then we may need our legal team to review it.